src/Monolog/Handler/AmqpHandler.php

```diff
@@ -117,6 +118,10 @@ public function handleBatch(array $records): void
             $record = $this->processRecord($record);
             $data = $this->getFormatter()->format($record);
 
+            if($data instanceof GelfMessage) {
+                $data = json_encode($data->toArray());
+            }
```

Comment:
This probably should be done in write() as well? Otherwise only handleBatch is supported, but not handling a single log record.

Reply:
But I do kinda wonder what the point is of this PR. Shouldn't GelfMessageFormatter be used with GelfHandler rather? Why use AMQP? And if you need AMQP why not simply use a JsonFormatter or LineFormatter?

Reply:
We prefer using the GelfMessageFormatter because of all the goodies of the implementation and tried to avoid implementing the specs a second time.

We have a local RabbitMQ broker per host and forward all log messages to a central server. So we don't talk to Graylog directly to keep the latency (`PHP->monolog->amqp->local RabbitMQ`) as low as possible.

Reply:
Ok fair enough, but anyway `write()` should still be fixed IMO for completeness

Reply:
Good point! I will have a look.

Reply:
Done.